src/libraries/System.Private.CoreLib/src/System/ByReference.cs

```diff
@@ -18,7 +18,7 @@ namespace System
 
 #pragma warning disable IDE0060
         [Intrinsic]
-        public ByReference(ref T value)
+        public ByReference(in T value)
```

Comment:
I know this isn't a public API, but this change makes me a bit nervous since it affects all internal callers. Seems like instead we should put an `Unsafe.AsRef` within the `ROS<T>` ctor.

Reply:
Really?  `in` here simply results in an extra [IsReadOnly] attribute on the parameter, whereas adding `Unsafe.AsRef` into the ROS constructor actually changes how much IL is there, what the JIT needs to do, could affect inlining, etc.  Isn't the latter actually riskier?

Reply:
`ByReference<T>` is equivalent of `ref T`. This change makes it logically possible to assign `readonly ref T` to `ref T` without any visible casts, that does not look right. As Levi said, this is not public API so it is not the end of the world.

Once ref fields are enabled in Roslyn (hopefully soon), we will be able to delete `ByReference<T>` and replace it with `ref T` or `readonly ref T`. In the meantime, we do not need to rock the boat.

Reply:
> In the meantime, we do not need to rock the boat.

Does "not rocking the boat" here mean:
- close this PR
- leave it as is
- change the ROS ctor to use Unsafe.AsRef as Levi suggested
?

Reply:
Keep `ByReference` as is, and change the ROS ctor to use Unsafe.AsRef as Levi suggested.

Reply:
I like this change otherwise.